model/src/pyrenew/process/rtperiodicdiff.py

```diff
@@ -0,0 +1,277 @@
+from jax.typing import ArrayLike
+from pyrenew.metaclass import Model, RandomVariable, _assert_sample_and_rtype
+from pyrenew.process.firstdifferencear import FirstDifferenceARProcess
+
+
+class RtPeriodicDiffProcessProcessSample(NamedTuple):
+    """
+    A container for holding the output from `process.RtWeeklyDiffProcess.sample()`.
+
+    Attributes
+    ----------
+    rt : ArrayLike
+        The sampled Rt.
+    """
+
+    rt: ArrayLike | None = None
+
+    def __repr__(self):
+        return f"RtPeriodicDiffProcessProcessSample(rt={self.rt})"
+
+
+class RtPeriodicDiffProcess(Model):
+    r"""
+    Periodic Rt with autoregressive difference.
+
+    Notes
+    -----
+    This class samples a periodic Rt with autoregressive difference. The
+    mathematical formulation is given by:
+
+    .. math::
+        \log[\mathcal{R}^\mathrm{u}(t_3)] \sim \mathrm{Normal}\left(\log[\mathcal{R}^\mathrm{u}(t_2)] \
+            + \beta \left(\log[\mathcal{R}^\mathrm{u}(t_2)] - \
+             \log[\mathcal{R}^\mathrm{u}(t_1)]\right), \sigma_r \right)
+
+    where :math:`\mathcal{R}^\mathrm{u}(t)` is the periodic reproduction number
+    at time :math:`t`, :math:`\beta` is the autoregressive parameter, and
+    :math:`\sigma_r` is the standard deviation of the noise.
+    """
+
+    def __init__(
+        self,
+        n_timepoints: int,
+        data_starts: int,
+        period_size: int,
+        log_rt_prior: RandomVariable,
+        autoreg: RandomVariable,
+        periodic_diff_sd: RandomVariable,
+        site_name: str = "rt_periodic_diff",
+    ) -> None:
+        """
+        Default constructor for RtPeriodicDiffProcess class.
+
+        Parameters
+        ----------
+        n_timepoints : int
+            Size of the returned sample.
+        data_starts : int
+            Relative point at which data starts, must be between 0 and
+            period_size - 1.
+        log_rt_prior : RandomVariable
+            Log Rt prior for the first two observations.
+        autoreg : RandomVariable
+            Autoregressive parameter.
+        periodic_diff_sd : RandomVariable
+            Standard deviation of the noise.
+        site_name : str, optional
+            Name of the site. Defaults to "rt_periodic_diff".
+
+        Returns
+        -------
+        None
+        """
+
+        self.validate(
+            n_timepoints,
+            data_starts,
+            log_rt_prior,
+            autoreg,
+            periodic_diff_sd,
+        )
+
+        self.n_timepoints = n_timepoints
+        self.period_size = period_size
+        self.data_starts = data_starts
+        self.n_periods = int(jnp.ceil(n_timepoints / period_size))
+        self.log_rt_prior = log_rt_prior
+        self.autoreg = autoreg
+        self.periodic_diff_sd = periodic_diff_sd
+        self.site_name = site_name
+
+        return None
+
+    @staticmethod
+    def validate(
+        n_timepoints: int,
+        data_starts: int,
+        prior: any,
```

Comment:
Minor note. On line 91, we have "log_rt_prior", whereas validate takes in `prior`. This makes sense because the prior doesn't necessarily have to be a `log_rt_prior`, it just has to pass what's in `validate`. However, the docstring on line 117 supports the interpretation of the prior as `log_rt_prior`, and the other parameters ingested by `validate` are unchanged in name, yet could be (e.g., `n_timepoints` to `ts`). Having consistency by having `validate` take in `log_rt_prior` (what is expected to be one) in place of `prior` might be good.

Reply:
Good point

Reply:
Addressed in 439367c